fatiando/gridder.py

```diff
@@ -439,3 +444,269 @@ def cut(x, y, scalars, area):
               if x[i] >= xmin and x[i] <= xmax
               and y[i] >= ymin and y[i] <= ymax]
     return [x[inside], y[inside], [s[inside] for s in scalars]]
+
+def pad_array(a, xy=None, npd=None, padtype='OddReflectionTaper'):
+    """
+    Return a padded array of arbitrary dimension.
+
+    The function takes an array of arbitrary dimension and pads it either to 
+    the dimensions given by the tuple npd, or to the next power of 2 if npd is
+    not given.  New coordinate vectors are computed for each dimension, if
+    provided.
+
+    An odd reflection with a cosine taper is the author's preferred method of
+    padding for Fourier operations.  The odd reflection optimally preserves
+    the frequency content while adding minimal sharp inflections.  The cosine
+    taper is a smooth function which also adds few sharp inflection points.
+
+    .. note:: Requires gridded data of the same dimension as desired
+        (i.e. no flattened arrays; use reshape).
+
+    .. note:: This function returns a deep copy of the original array.
+
+    .. warning:: This function returns the coordinate vectors as a list of 
+        arrays.  This means that even for a 1D input array, the returned 
+        vector will be in a list of one element.
+
+    Parameters:
+
+    * a : numpy array
+        numpy array (N-D) to be padded
+    * xy : N-D array (optional)
+        [MxN] array where M is the number of observation points and N is the 
+        dimension.  This is effectively a concatenated xp,yp, etc...
+    * npd : tuple (optional)
+        Optional tuple containing the total number of desired elements in each
+        dimension
+    * padtype : string (optional)
+        String describing what to pad the new values with. Options:
+
+        [ OddReflectionTaper | OddReflection | Reflection | value | LinTaper
+        | edge | mean ]
+            OddReflectionTaper - Generates odd reflection then tapers to the 
+            mean using a cosine function (Default)
+
+            OddReflection - Pads with the odd reflection, with no taper
+
+            Reflection - Pads with simple reflection
+
+            LinTaper - Linearly tapers to the mean
+
+            value - numeric value
+
+            edge - uses the edge value as a constant pad
+
+            mean - uses the mean of the vector along each axis
+
+    Returns:
+
+    * ap : numpy array
+        Padded array. The array core is a deep copy of the original array
+    * cp : list
+        List of coordinate arrays containing the extrapolated coordinate values
+        Returns None if xy not provided
+    * nps : list
+        List of tuples containing the number of elements padded onto each 
+        dimension.
+
+    """
+
+    # Test to make sure padtype is valid
+    padopts = ['','oddreflectiontaper','oddreflection', 'reflection',
+            'lintaper', 'edge', 'value', 'mean']
```

Comment:
> It's a luxury--it would be encoded in the padded array, although if you had an odd number of padding elements, you would need to know which side had the extra element. You could find that out by comparing your coordinate vectors, or by knowing which side the extra element is padded on. I personally think it makes life easier, if a tad more messy.

You're completely right! I hadn't thought of odd number padding. That's perfect the way it is then. 

Maybe you could add a small example of how that argument looks? That might help clarify. It could even be a doctest like [this one](https://github.com/fatiando/fatiando/blob/master/fatiando/gridder.py#L141). That would help documenting and testing at the same time.